### DISCUSSION TITLE
Send or forward email via http api to local on premise email server

### DISCUSSION THREAD

**Discussion by geoidwalker** [postalserver/postal#2560] [Help with using Postal] [answered]
Good morning,
We have an on-premise exchange server and an external spam filter (proofpoint). Email sent between staff members (our domain) that is sent from outside our network goes through proofpoint. However email that is sent from a computer on the internal network goes direct to our exchange server. 

I'd like to have our software use an email api to send directly to our on premise Exchange server. With postal, I can successfully send email but it goes through our external proofpoint server. Is there a way to configure postal to send directly to the on premise mail server using the http api? I suspect its something to do with the way exchange and active sync is set up so it may be I need a different solution. Essentially, we have software (web applications) that needs to send a lot of emails but only to our staff and only on the local network. 

Any help would be greatly appreciated. 

G

**Comment by willpower232** [comment, marked as answer]
You can configure Postal to use a local nameserver to look up DNS records

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/config/postal.defaults.yml#L17

Depending on how your email domain is set up, either this will be enough or you could use the hosts file on the Postal server to point directly at the Exchange server or something along those lines.

**Reply by geoidwalker** [reply]
Thanks very much for the reply. I added the config you noted (set to true). 

Editing the /etc/hosts file didn't have any effect but I had a clue from the error message I got which was 

Permanent SMTP delivery error when sending to mx-proofpointDNS (externalIP) Recipient added to suppression list (too many hard fails).

I made sure the recipient address is removed from the suppression list by dropping it from the database. 

I added all the necessary DNS records to our local nameserver - adding the missing A record to point to my postal server was the missing link. I have 4 green check marks now. 
However, we have MX records already pointing to proofpoint. After setting the priority for mx.postal.mydomain to 10 and the other two MX records to 20 and 30 the email api started working. I had to add a route of course to the local exchange server. 

I tested sending mail to internal and external addresses (using outlook) - and I tested the postal api..everything is working - but I'm a bit worried that since mx.postal is priority 10 that outlook users might start seeing email fail as it may try to send email thru postal. I'm actually not clear why when I send email normally thru outlook it doesn't go thru postal as that record has the higher priority. I'll observe and see if there are any issues. 

I'm also looking at setting up a separate domain like myemail@postal.mydomain.com - but I suppose then I need to configure that in Exchange. I'll play with it a bit more on Monday but I'm definitely much farther than I was! Thanks again for the help.

**Reply by willpower232** [reply]
if your email address is whatever@example.com
then you have mx records referencing office.example.com
and office.example.com is public IP 1.2.3.4
and your exchange server is internal IP 5.6.7.8

then your hosts file for the Postal server should say that office.example.com is internal IP 5.6.7.8 so Postal doesn't use the external network at all

if that didn't work too well, you could use dnsmasq as a docker container to talk DNS records more officially to provide the altered record to Postal.
